naucse/markdown_util.py

```diff
@@ -67,6 +67,20 @@ def ansi_convert(code):
     return ansi_convertor.convert(replaced, full=False)
 
 
+def style_space_after_prompt(html):
+    return re.sub(r'<span class="gp">([^<]*[^<\s])</span>(\s)',
+                  r'<span class="gp">\1\2</span>',
+                  html)
+
+
+def get_lexer_by_name(lang):
+    lexer = pygments.lexers.get_lexer_by_name(lang)
+    if lang == 'doscon':
+        # workaround for pygments #1380
+        lexer._ps1rgx = r'^([^>]*>)(.*\n?)'
```

Comment:
This lexes the output of `dir` incorrectly, if there are subdirectories:
```
> dir
 Directory of C:\Users\helena
05/08/2014 07:28 PM <DIR>  Applications
05/08/2014 07:28 PM <DIR>  Desktop
05/08/2014 07:28 PM <DIR>  Downloads
05/08/2014 07:28 PM <DIR>  Music
…
```

Of course `console` wasn't ideal for Windows shell. But if we're hacking up workarounds, I think it the result should work with our examples.

Perhaps it would be better to not try to perfect a general-purpose `doscon`, but make a custom "DOS shell with optional venv" using something like `r'^((\([_\w]+\))]?>\s?)(.*\n?)'`?